subtrack.MAUI/Utilities/DateTimeExtensions.cs

```diff
@@ -1,19 +1,18 @@
 ﻿using subtrack.MAUI.Services.Abstractions;
 
 namespace subtrack.MAUI.Utilities;
+
 public static class DateTimeExtensions
 {
-    public static TimeSpan TimeRemainingFromToday(this DateTime date, IDateProvider dateTimeProvider)
-    {
-        return date.Subtract(dateTimeProvider.Today);
-    }
+    public static TimeSpan TimeRemainingFromToday(this DateTime date, IDateProvider dateTimeProvider) => date.Subtract(dateTimeProvider.Today);
+
+    public static bool IsPastDate(this DateTime date, IDateProvider dateTimeProvider) => date < dateTimeProvider.Today;
+
+    public static DateTime AddWeeks(this DateTime date, int weeks) => date.AddDays(weeks * 7);
+
+    public static string MonthName(this DateTime date) => date.ToString("MMMM");
+
+    public static DateTime LastDayOfMonth(this DateTime date) => new(date.Year, date.Month, DateTime.DaysInMonth(date.Year, date.Month));
```

Comment:
This is nice, maybe change the name to be LastDayOfMonthDate since otherwise one could expect an int to be returned